src/smt/mod.rs

```diff
@@ -42,33 +33,24 @@ pub struct SmtCtx<'ctx> {
     lists: RefCell<HashMap<TyKind, Rc<ListFactory<'ctx>>>>,
     lits: RefCell<Vec<(Sort<'ctx>, LitDecl<'ctx>)>>,
     uninterpreteds: Uninterpreteds<'ctx>,
-    limited_function_encoding: LimitedFunctionEncoding<'ctx>,
-    options: SmtCtxOptions,
+    function_encoding: FunctionEncoding<'ctx>,
+    lit_wrap: bool,
 }
 
 impl<'ctx> SmtCtx<'ctx> {
-    pub fn new(ctx: &'ctx Context, tcx: &'ctx TyCtx, options: SmtCtxOptions) -> Self {
+    pub fn new(
+        ctx: &'ctx Context,
+        tcx: &'ctx TyCtx,
+        function_encoding: FunctionEncoding<'ctx>,
+    ) -> Self {
         let domains: Vec<_> = tcx.domains_owned();
         // disable lit-wrapping if there are no limited functions that can profit from it
-        let lit_wrap = options.lit_wrap
-            && options.use_limited_functions
+        let lit_wrap = function_encoding.needs_lit_wrapping()
```

Comment:
None of the encoding *kinds* really *need* any kind of lit wrapping, right? Whether we use fixed/variable fuel is independent of lit wrapping?